File: benchmarks/matrix_mul.c
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

static int64_t* read(int64_t n) {
  int64_t k = n * n;
  int64_t* A = malloc(k * sizeof(*A));
  int64_t i = 0;
  while (i < k) {
    int64_t t = 0;
    scanf("%" PRId64, &t);
    A[i] = t;
    i = i + 1;
  }
  return A;
}

static void write(int64_t* M, int64_t n) {
  int64_t k = n * n;
  int64_t i = 0;
  while (i < k) {
    printf("%" PRId64 "\n", M[i]);
    i = i + 1;
  }
}

static int64_t* mul(int64_t* A, int64_t* B, int64_t n) {
  int64_t size = n * n;
  int64_t* C = malloc(size * sizeof(*A));
  int64_t i = 0;
  while (i < n) {
    int64_t j = 0;
    while (j < n) {
      int64_t sum = 0;
      int64_t k = 0;
      while (k < n) {
        sum = sum + A[i * n + k] * B[k * n + j];
        k = k + 1;
      }
      C[i * n + j] = sum;
      j = j + 1;
    }
    i = i + 1;
  }
  return C;
}

int main() {
  int64_t n = 0;
  scanf("%" PRId64, &n);
  int64_t* A = read(n);
  int64_t* B = read(n);
  int64_t* C = mul(A, B, n);
  write(C, n);
}

File: benchmarks/matrix_mul.java
import java.util.Scanner;

public class matrix_mul {
  private static long[] read(Scanner in, int n) {
    int k = n * n;
    long[] A = new long[k];
    int i = 0;
    while (i < k) {
      long t = in.nextLong();
      A[i] = t;
      i = i + 1;
    }
    return A;
  }

  private static void write(long[] M, int n) {
    int k = n * n;
    int i = 0;
    while (i < k) {
      System.out.println(M[i]);
      i = i + 1;
    }
  }

  private static long[] mul(long[] A, long[] B, int n) {
    int size = n * n;
    long[] C = new long[size];
    int i = 0;
    while (i < n) {
      int j = 0;
      while (j < n) {
        long sum = 0;
        int k = 0;
        while (k < n) {
          sum = sum + A[i * n + k] * B[k * n + j];
          k = k + 1;
        }
        C[i * n + j] = sum;
        j = j + 1;
      }
      i = i + 1;
    }
    return C;
  }

  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    int n = in.nextInt();
    long[] A = read(in, n);
    long[] B = read(in, n);
    long[] C = mul(A, B, n);
    write(C, n);
    in.close();
  }
}
